scripts/loggeranalyzer.py

```diff
@@ -36,191 +41,379 @@
+    re.IGNORECASE,
+)
 
-def jsonl_formatter(errors: List[Optional[Dict]]) -> List[str]:
-    """Imprime linha a linha da lista de entrada, convertendo o conteúdo para
-    JSON. É esperado então que uma lista de dados seja transformada no formato
-    JSONL."""
-
-    result = []
-    for error in errors:
-        result.append(json.dumps(error))
-
-    return result
-
-
-FORMATTERS = {"jsonl": jsonl_formatter}
-
-
-def output_lines_to(lines: List[str], stream: TextIOWrapper) -> None:
-    """Escreve resultado do parser em um arquivo caminho determinado.
-    O stream de dados deve implementar a interface TextIOWrapper"""
 
-    for line in lines:
-        stream.write(line)
-        stream.write("\n")
+class ErrorEnum(Enum):
+    """Enumerador para agrupar nomes de erros utilizados na saída.
 
+    Classe enumeradora.
 
-class ErrorEnum(Enum):
-    """Enum para agrupar nomes de erros utilizados
-    na saída do script"""
+    Atributos:
+        Não há atributos.
+    """
 
     RESOURCE_NOT_FOUND = "resource-not-found"
     NOT_UPDATED = "xml-not-update"
     XML_NOT_FOUND = "xml-not-found"
     MISSING_METADATA = "missing-metadata"
+    MISSING_MANIFEST = "missing-manifest"
+    XML_PARSER_ERROR = "xml-parser-error"
+    BUNDLE_NOT_FOUND = "bundle-not-found"
+    ISSN_NOT_FOUND = "issn-not-found"
+    PACKAGE_NOT_FOUND = "package-not-found"
+
+
+PACK_FROM_SITE_PARSERS_PARAMS = [
+    {
+        "regex": ASSET_NOT_FOUND_REGEX,
+        "error": ErrorEnum.RESOURCE_NOT_FOUND,
+        "group": "renditions",
+    },
+    {
+        "regex": RENDITION_NOT_FOUND_REGEX,
+        "error": ErrorEnum.RESOURCE_NOT_FOUND,
+        "group": "renditions",
+    },
+    {"regex": XML_NOT_UPDATED_REGEX, "error": ErrorEnum.NOT_UPDATED},
+    {"regex": XML_NOT_FOUND_REGEX, "error": ErrorEnum.XML_NOT_FOUND},
+    {"regex": XML_MISSING_METADATA_REGEX, "error": ErrorEnum.MISSING_METADATA},
+]
 
 
-def general_parser(
-    line: str, regex: re.Pattern, error: ErrorEnum, group: str = None
-) -> Optional[Dict]:
-    """Parser de formato geral utilizado para verificar padrões por meio
-    de regex. Retorna o tipo de formato especificado na chamada."""
-
-    match = regex.match(line)
-
-    if match is None:
-        return None
-
-    return dict(match.groupdict(), **{"error": error.value, "group": group})
+IMPORT_PARSERS_PARAMS = [
+    {"regex": MANIFEST_NOT_FOUND, "error": ErrorEnum.MISSING_MANIFEST},
+    {"regex": XML_NOT_INTO_PACKAGE, "error": ErrorEnum.XML_NOT_FOUND},
+    {"regex": XML_PARSER_ERROR, "error": ErrorEnum.XML_PARSER_ERROR},
+    {"regex": BUNDLE_NOT_FOUND, "error": ErrorEnum.BUNDLE_NOT_FOUND},
+    {"regex": ISSN_NOT_FOUND, "error": ErrorEnum.ISSN_NOT_FOUND},
+    {"regex": PACKAGE_NOT_FOUND, "error": ErrorEnum.PACKAGE_NOT_FOUND},
+]
 
 
-parse_asset_error = functools.partial(
-    general_parser,
-    regex=ASSET_NOT_FOUND_REGEX,
-    error=ErrorEnum.RESOURCE_NOT_FOUND,
-    group="assets",
+class LoggerAnalyzer(object):
```

Comment:
@jamilatta, porque foi necessário introduzir essa classe ao invés de seguir o padrão adotado pelo script? Note que antes tudo era feito através de composição de função. Cada grupo de funções / parsers podiam implementar a interface desejada.

Agora nós passamos a utilizar dicionários que são consumidos pelo método `parser` que espera algo bem definido com as chaves de `regex`, `group` e `error`.

Se a ideia for manter a estrutura de classes, eu sugiro você criar uma interface que determina os métodos. Assim cada `parser` pode implementar sua lógica interna.

Reply:
@joffilyfe 

Irei realizar esse ajuste!